compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/IsolationAnalyzer.java

```diff
@@ -2383,6 +2386,22 @@ private void analyzeArgIsolatedness(BLangInvocation invocationExpr, List<BLangEx
         analyzeRestArgsForRestParam(invocationExpr, restArgs, symbol, expectsIsolation);
     }
 
+    private BTupleType getRepresentativeTupleTypeForRemainingArgs(int paramCount, int reqArgCount,
+                                                                  BArrayType arrayType) {
+        int remReqArgCount = paramCount - reqArgCount;
+        List<BType> memTypes = new ArrayList<>(remReqArgCount);
+        BType eType = arrayType.eType;
+        for (int i = 0; i < remReqArgCount; i++) {
+            memTypes.add(eType);
+        }
```

Comment:
We can refactor this in the following manner.
```
List<BType> memTypes = new ArrayList<>(Collections.nCopies(remReqArgCount, eType));
```

Reply:
This creates an immutable list, right? I'm not sure it's safe to due atm since the compiler isn't written assuming immutable member lists (in the cyclic case we even add, although it won't be cyclic here).

Reply:
Also with the tuple change it isn't a copy. We need to create new tuple members, will be updating the PR now.

Reply:
`Collections.nCopies` returns an immutable list, but it can be used to initialize a new mutable ArrayList instance.